test/BumpVersionSpec.hs

```diff
@@ -53,70 +59,118 @@ setOverrides overrides = do
+              && _svPatch newVersion == _svPatch initialVersion + 1
+
+      prop "parseVersion . unparseVersion == id" $
+        forAll randomSemVer $ \v -> parseVersion (unparseVersion v) == Right v
+
+  describe "unit tests" $ do
+
+    describe "parseVersionBump" $ do
+
+      it "should parse 'major'" $
+        parseVersionBump "major" `shouldBe` Just Major
+
+      it "should parse 'minor'" $
+        parseVersionBump "minor" `shouldBe` Just Minor
 
-  describe "spago bump-version" $ do
+      it "should parse 'patch'" $
+        parseVersionBump "patch" `shouldBe` Just Patch
 
-    it "Spago should complain when no git repo exists" $ do
+      it "should parse version starting with 'v'" $
+        parseVersionBump "v1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-      spago ["bump-version", "minor"] >>= shouldBeFailureInfix
-        "Your git working tree is dirty. Please commit or stash your changes first"
+      it "should parse version not starting with 'v'" $
+        parseVersionBump "1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-    before_ (initGitTag "v1.2.3") $ it "Spago should not make a tag when not passing --no-dry-run" $ do
+      -- TODO is this desired behavior, or should we just drop ONE 'v'? I'd agree it's edge case, but still :-)
+      it "should drop multiple 'v's from the beginning" $
+        parseVersionBump "vvvvvvvv1.2.3" `shouldBe` Just (Exact (SemVer 1 2 3 [] []))
 
-      spago ["bump-version", "minor"] >>= shouldBeSuccess
-      getHighestTag >>= (`shouldBe` Just "v1.2.3")
+      -- TODO should this work or should we strip these in parser implementation?
+      it "should parse versions with PREREL and META tags" $
+        parseVersionBump "1.2.3-r1+git123" `shouldBe` Just (Exact (SemVer 1 2 3 [[Str "r",Digits 1]] [[Str "git", Digits 123]]))
 
-    before_ (initGitTag "not-a-version") $ it "Spago should use v0.0.0 as initial version" $ do
+      it "should not parse version which is not semantic" $ do
+        parseVersionBump "" `shouldBe` Nothing
+        parseVersionBump "1" `shouldBe` Nothing
+        parseVersionBump "1.2" `shouldBe` Nothing
+        parseVersionBump "1.2.3.4" `shouldBe` Nothing
 
-      spago ["bump-version", "--no-dry-run", "patch"] >>= shouldBeSuccess
-      getHighestTag >>= (`shouldBe` Just "v0.0.1")
+  around_ setup $ do
+    describe "end to end tests" $ do
 
-    before_ (initGitTag "v1.3.4") $ it "Spago should bump patch version" $ do
+      it "Spago should complain when no git repo exists" $ do
 
-      spago ["bump-version", "--no-dry-run", "patch"] >>= shouldBeSuccess
-      getHighestTag >>= (`shouldBe` Just "v1.3.5")
+        spago ["bump-version", "minor"] >>= shouldBeFailureInfix
+          "Your git working tree is dirty. Please commit or stash your changes first"
 
-    before_ (initGitTag "v1.3.4") $ it "Spago should bump minor version" $ do
+      before_ (initGitTag "v1.2.3") $ it "Spago should only make a tag with `--no-dry-run`" $ do
```

Comment:
The basic idea was to replace [these tests](https://github.com/spacchetti/spago/blob/52ddd50747bd139474008780cc612396a4b5477b/test/BumpVersionSpec.hs#L73-L94) with just one - they all test the same path, just differ in git tag version, which I think is adequately tested in the unit/property tests above.

Reply:
:+1: